sktime/annotation/base/_base.py

```diff
@@ -314,3 +345,231 @@ def _update(self, X, Y=None):
         self._fit(self._X, self._Y)
 
         return self
+
+    def predict_segments(self, X):
+        """Predict segments on test/deployment data.
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Data to annotate, time series.
+
+        Returns
+        -------
+        Y : np.ndarray
+            2D array containing the segments for X. The first column contains the labels
+            of the segments, the second column contains the starting indexes of the
+            segments.
+        """
+        if self.task == "anomaly_detection":
+            raise RuntimeError(
+                "Anomaly detection annotators should not be used for segmentation."
+            )
+        self.check_is_fitted()
+        X = check_series(X)
+
+        if self.task == "change_point_detection":
+            return self.segments_to_change_points(self.predict_points(X))
+        elif self.task == "segmentation":
+            return self._predict_segments(X)
+
+    def predict_points(self, X):
+        """Predict changepoints/anomalies on test/deployment data.
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Data to annotate, time series.
+
+        Returns
+        -------
+        Y : np.ndarray
+            1D array containing the indexes of the changepoints/anomalies in X.
```

Comment:
as above, I feel we should account for the option to return labels too.

I have a small preference for `pd.DataFrame`, by reasoning of consistency.

Reply:
Okay, I have had a go at making the switch to pandas dataframes and series.